### ISSUE TITLE
TypeScript: Confused about semicolons with in nested interfaces

### ISSUE THREAD

**Issue by prencher** [prettier/prettier#1879]
Prettier replaces the semicolon with a comma in the following input:
```ts
export interface Edge {
  cursor: {};
  node: {
    id: {};
  };
}
```

Output with `--trailing-comma all`:
```ts
export interface Edge {
  cursor: {};
  node: {
    id: {},
  };
}
```
Expected: No change.

Without `--trailing-comma`:
```ts
export interface Edge {
  cursor: {};
  node: {
    id: {}
  };
}
```
Expected: All semicolons being stripped.

Prettier seems to think `node` is an object, rather than a nested interface. Technically valid, but incorrect. 😄 

Corner case using commas results in the same outputs as the above outputs with semicolon inputs:
```ts
export interface Edge {
  cursor: {},
  node: {
    id: {},
  },
}
```

**Comment by vjeux** [comment]
@azz, do you have any idea here?

**Comment by kamek-pf** [comment]
Also seeing inconsistent behavior with that kind of interface.

**Comment by vjeux** [comment]
I just published 1.4.1 which fixes it!
